### ISSUE TITLE
Unable to compile pi_sonar

### ISSUE THREAD

**Issue by JGUO006** [UbiquityRobotics/pi_sonar#15]
Hi, 

I am using Ubuntu 16.04 with 2020-11-07 ubiquity-xenial-lxde image running on Raspberry Pi 4.

I followed these steps to handle the Raspberry Pi 4 Issues With Sonar board but encountered a cmake error.

Could you please help with it?

![image](https://user-images.githubusercontent.com/78726874/109608191-76805500-7b64-11eb-9a4b-81f257ca0bc2.png)

![image](https://user-images.githubusercontent.com/78726874/109608467-e393ea80-7b64-11eb-88fd-f0cfd156c099.png)

**Comment by JanezCim** [comment]
Interesting, i just did the same yesterday and it worked on a RPI4.
Can you try:
`rosdep install --from-paths src --ignore-src -r -y`
and then recompile?

**Comment by JGUO006** [comment]
Hi, @JanezCim 

I tried this command and same compiling error occurred. 

I remembered previously I updated cmake version to 3.20.0. 

I don't know whether this could affect this compiling but other packages (demos, rplidar_ros) under this catkin_ws folder can be successfully compiled. 

![image](https://user-images.githubusercontent.com/78726874/109740931-11ca0680-7c07-11eb-9c0b-312f5fa46109.png)

**Comment by JanezCim** [comment]
Okay got around to actually doing this again with fresh `2020-11-07-ubiquity-xenial-lxde-raspberry-pi.img`. It compiled fine using these commands:
```
rosnode kill -a
cd ~/catkin_ws/src/
git clone https://github.com/UbiquityRobotics/pi_sonar
cd pi_sonar/
git checkout daemon_pigpio
cd ~/catkin_ws/
catkin_make -j2
```

Maybe redownload and reflash the image as well and try again

**Comment by JGUO006** [comment]
@JanezCim I have downloaded and flashed the fresh image 2020-11-07-ubiquity-xenial-lxde-raspberry-pi.img.

It complied fine by following the steps above (with the original built-in CMake version 3.5.1). Thanks for your reply and help!

Previously I manually upgraded CMake version to 3.20.0 and that probably caused this compiling issue.

![image](https://user-images.githubusercontent.com/78726874/110299874-340dbb00-8031-11eb-9b82-869bf1cebf78.png)
